[Build] Update Gradle to 8.12.1 and AGP to 8.8.1

ParaskP7 opened the pull request:
Issue: [apps-infra-plans#174](https://github.com/Automattic/apps-infra-plans/issues/174)
Depends On (`for local builds`): [GBKit#92](https://github.com/wordpress-mobile/GutenbergKit/pull/92)

## Description

This PR updates Gradle to [8.12.1](https://docs.gradle.org/8.12.1/release-notes.html) and AGP to [8.8.1](https://developer.android.com/build/releases/gradle-plugin#8-8-1) ([context](https://github.com/wordpress-mobile/WordPress-Android/pull/21515#issuecomment-2545405811)).

-----

### Review

During review, please take a closer look at the below commits:

- Lint Suppression List:
    1. [Suppress credential manager misuse lint warning on login mdl](https://github.com/wordpress-mobile/WordPress-Android/commit/3559aa373ebf4214bf0cc6f1dda1eb728b63dc96)
    2. [Suppress set text i18n lint warnings on wordpress module](https://github.com/wordpress-mobile/WordPress-Android/commit/b850ea17e4794319183af4ba1bc2dec0a76e674c)

-----

## To Test:

<!-- Test instructions per dependency update: https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/docs/test_instructions_per_dependency_update.md -->


### Testing Steps

1. Verify that all the CI checks are successful.
2. Smoke test both apps, Jetpack and WordPress, without composite builds, and verify everything is working as expected.
3. Smoke test both apps again, Jetpack and WordPress, but this time, using the composite builds, focusing solely on [GBK](https://github.com/wordpress-mobile/GutenbergKit) (following the instructions in [local-builds.gradle-example](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/local-builds.gradle-example) file).
4. Try to test for build time increases both, locally and on CI.
    - For example, all main jobs of this [initial build](https://buildkite.com/automattic/wordpress-android/builds/21173) on CI seem to take a bit longer than usual:
        - [Jetpack App](https://buildkite.com/automattic/wordpress-android/builds/21173#0194faaa-8630-4c32-9f95-8fbabcf4d5d3): `20m 30s` ([scan](https://gradle.com/s/wjxkci2ohdgj4))
        - [WordPress App](https://buildkite.com/automattic/wordpress-android/builds/21173#0194faaa-862f-4209-9624-ed6d7822b9aa): `19m 34s` ([scan](https://scans.gradle.com/s/dsdxjfzr65x2m))
        - [Jetpack Lint](https://buildkite.com/automattic/wordpress-android/builds/21173#0194faaa-8626-458f-83c4-7ebe608e9cd4): `24m 12s` ([scan](https://scans.gradle.com/s/6tgvzwv46aioe))
        - [WordPress Lint](https://buildkite.com/automattic/wordpress-android/builds/21173#0194faaa-8625-4eb7-873b-97e9b2ee6206): `27m 20s` ([scan](https://scans.gradle.com/s/afzcpi5osq6u2))
        - [WordPress Unit Tests](https://buildkite.com/automattic/wordpress-android/builds/21173#0194faaa-8629-41cd-a4a1-d29c7a9a5c32): `20m 15s` ([scan](https://scans.gradle.com/s/kq6u6pmvnolsk))
        - [Processors Unit Tests](https://buildkite.com/automattic/wordpress-android/builds/21173#0194faaa-862a-4a3e-b7c3-d96131d198b3): `5m 5s` ([scan](https://scans.gradle.com/s/3iubmuurlh2ls))
        - [Image Editor Unit Tests](https://buildkite.com/automattic/wordpress-android/builds/21173#0194faaa-862b-4483-bd5f-63a914a9a698): `5m 43s` ([scan](https://scans.gradle.com/s/mxa6wjn3xakf6))
        - [FluxC Unit Tests](https://buildkite.com/automattic/wordpress-android/builds/21173#0194faaa-862c-49bc-a69c-292ed12ffa0d): `13m 20s` ([scan](https://scans.gradle.com/s/52c2ufh5wx7ug))
        - [Login Unit Tests](https://buildkite.com/automattic/wordpress-android/builds/21173#0194faaa-862d-460a-9b37-617e2569bd7c): `10 59s` ([scan](https://scans.gradle.com/s/dza46y3kakyyk))
    - This is because dependency resolution and network activity skyrocketed, from a few seconds, to minutes again, from just a couple of hundred network requests to 3-4K of those, effectively ignoring the dependency cache altogether. This is expected due to the bump in Gradle versions, from `8.8.1` to `8.12.1`. This diff will go back to normal as soon as this PR get merged and a fresh copy of dependency cache is saved.

        - FYI: Reading the [docs](https://docs.gradle.org/current/userguide/dependency_caching.html#sec:cache-copy) you'll find out that `metadata-2.106` is used with `Gradle 8.2 to Gradle 8.10.2` and `metadata-2.107` is used with `Gradle 8.11 and above`. We are indeed creating/using a new metadata cache since Gradle is now updated to `8.12.1`:
        > Note that creating the cache and consuming it should be done using compatible Gradle version, as shown in the table below. Otherwise, the build might still require some interactions with remote repositories to complete missing information, which might be available in a different version. If multiple incompatible Gradle versions are in play, all should be used when seeding the cache.
    - Triggering [another build](https://buildkite.com/automattic/wordpress-android/builds/21174) on CI seems to make things much better, most probably because we now have a build cache available from the [initial build](https://buildkite.com/automattic/wordpress-android/builds/21173). But, and as expected, the dependency cache problem still persists ([scan](https://scans.gradle.com/s/y7mwynpz7lw6e)).
5. Locally, in addition to the debug variants, assemble/lint the release variants, and verify everything is working as expected.
    - Run: `./gradlew assembleJetpackVanillaRelease` or/and `./gradlew assembleWordpressVanillaRelease`
    - Run: `./gradlew lintJetpackVanillaRelease` or/and `./gradlew lintWordpressVanillaRelease`
6. ~~Check [build environment changes](https://github.com/wordpress-mobile/WordPress-Android/pull/21680#issuecomment-2653998105).~~ Unfortunately, this is not possible due to this `Content exceeds 65400 characters.` problem. Good news are, that we are aware of this problem and plan to fix it, or else, work around it (see [apps-infra-plans#182](https://github.com/Automattic/apps-infra-plans/issues/182)).

---

### Merge Instructions

- [x] Update the [GBK](https://github.com/wordpress-mobile/GutenbergKit) library to PR hash (this is done for testing purposes).
    - [x] [[Build] Update Gradle to 8.12.1 and AGP to 8.8.1 #92](https://github.com/wordpress-mobile/GutenbergKit/pull/92) -> 619e7e3705d9c75f7ef3bdde84172e8a99699772
- [x] Mark PR as `Ready for review`.
    - [x] [Comment](https://github.com/wordpress-mobile/WordPress-Android/pull/21680#issuecomment-2656739991)
- [x] Wait for PR to get approved.
- [x] Merge the [GBK](https://github.com/wordpress-mobile/GutenbergKit) library PR into its `trunk` (after them being code reviewed and approved).
    - [x] [[Build] Update Gradle to 8.12.1 and AGP to 8.8.1 #92](https://github.com/wordpress-mobile/GutenbergKit/pull/92) -> b9e2e1b329678bd848d94bd998e41c584a7e9872
- [x] Update the [GBK](https://github.com/wordpress-mobile/GutenbergKit) library reference to `trunk` hash or tag.
- [x] Remove `[Status] Not Ready for Merge` label.
- [x] Merge this PR.

---

### Post Merge Instructions

- [x] ~~Triggering [schedules/dependency-cache.yml](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/.buildkite/schedules/dependency-cache.yml) manually via [this](https://buildkite.com/automattic/wordpress-android) `New Build` button to generate & save an update dependency cache.~~ -> [Done](https://buildkite.com/automattic/wordpress-android/builds/21210)
- [x] ~~Look at newly started builds and verify that dependency cache restoration works as expected, with dependency resolution and network activity dropping to expected levels (just a couple of seconds for dependency resolution, close to zero of network activity).~~ -> [Done](https://buildkite.com/automattic/wordpress-android/builds/21211) (see [Lint Jetpack](https://buildkite.com/automattic/wordpress-android/builds/21211#019518bf-f269-494c-a9ee-f5c8f3c052cf) [scan](https://scans.gradle.com/s/dp7kudgaflcfk) with [few seconds of dependency resolution](https://scans.gradle.com/s/dp7kudgaflcfk/performance/dependency-resolution) and [zero network activity](https://scans.gradle.com/s/dp7kudgaflcfk/performance/network-activity), as expected)
- [ ] ~~Run another set of testing/benchmarking for CC on CI (based on #21659) and re-measuring whether disabling CC on CI actually helps reduce build times (or not), and if so by how much.~~ --> [Done](https://github.com/wordpress-mobile/WordPress-Android/pull/21690)

-----

## Regression Notes

1. Potential unintended areas of impact

    - CI builds failing or slower than expected.

3. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To Test` section above.

5. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones): `N/A`

ParaskP7 committed:
Build: Upgrade agp to 8.6.1

Release Notes: https://developer.android.com/build/releases/
past-releases/agp-8-6-0-release-notes

ParaskP7 committed:
Build: Upgrade gradle to 8.9

Release Notes: https://docs.gradle.org/8.9/release-notes.html

Command: ./gradlew wrapper --gradle-version=8.9
--distribution-type=all --gradle-distribution-sha256-sum=
258e722ec21e955201e31447b0aed14201765a3bfbae296a46cf60b70e66db70

ParaskP7 committed:
Build: Upgrade agp to 8.7.3

Release Notes: https://developer.android.com/build/releases/
past-releases/agp-8-7-0-release-notes

ParaskP7 committed:
Build: Upgrade gradle to 8.10.2

Release Notes: https://docs.gradle.org/8.10.2/release-notes.html

Command: ./gradlew wrapper --gradle-version=8.10.2
--distribution-type=all --gradle-distribution-sha256-sum=
2ab88d6de2c23e6adae7363ae6e29cbdd2a709e992929b48b6530fd0c7133bd6

ParaskP7 committed:
Build: Upgrade agp to 8.8.0

Release Notes: https://developer.android.com/build/releases/
gradle-plugin

ParaskP7 committed:
Build: Upgrade gradle to 8.11.1

Release Notes: https://docs.gradle.org/8.11.1/release-notes.html

Command: ./gradlew wrapper --gradle-version=8.11.1
--distribution-type=all --gradle-distribution-sha256-sum=
89d4e70e4e84e2d2dfbb63e4daa53e21b25017cc70c37e4eea31ee51fb15098a

ParaskP7 committed:
Build: Upgrade gradle to 8.12.1

Release Notes: https://docs.gradle.org/8.12.1/release-notes.html

Command: ./gradlew wrapper --gradle-version=8.12.1
--distribution-type=all --gradle-distribution-sha256-sum=
296742a352f0b20ec14b143fb684965ad66086c7810b7b255dee216670716175

ParaskP7 committed:
Analysis: Suppress credential manager misuse lint warning on login mdl

This Lint warning can be suppressed because 'NoCredentialException' is
actually a subclass of 'GetCredentialException', which means that it is
already caught.

Warning Message: "Call to CredentialManager.getCredential without use of
NoCredentialException"

Explanation: "When calling CredentialManager.getCredential or
CredentialManager.getCredentialAsync, you should handle
NoCredentialException somewhere in your project.

More info: https://developer.android.com/identity/sign-in/
credential-manager#handle-exceptions"

ParaskP7 committed:
Analysis: Suppress set text i18n lint warnings on wordpress module

I18n is not actually needed on these specific cases.

Warning Messages: "Number formatting does not take into account locale
settings. Consider using String.format instead."

Explanation: "When calling TextView#setText
* Never call Number#toString() to format numbers; it will not handle
fraction separators and locale-specific digits properly. Consider using
String#format with proper format specifications (%d or %f) instead.
* Do not pass a string literal (e.g. "Hello") to display text. Hardcoded
text can not be properly translated to other languages. Consider using
Android resource strings instead.
* Do not build messages by concatenating text chunks. Such messages can
not be properly translated.

More info: https://developer.android.com/guide/topics/resources/
localization.html"

dangermattic commented:
<!--
  0 Errors
  1 Warning: View files have been modified,...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

nbradbury commented on WordPress/src/main/java/org/wordpress/android/ui/history/RevisionItemViewHolder.kt:
```diff
@@ -46,13 +47,15 @@ class RevisionItemViewHolder(
             diffLayout.visibility = View.VISIBLE
 
             if (boundRevision.totalAdditions > 0) {
+                @SuppressLint("SetTextI18n")
```
@ParaskP7 I realize this is still a draft, but this suppression caught my eye and I wanted to point out we can remove it if we instead catch `NoCredentialException`below.

nbradbury commented on libs/login/src/main/java/org/wordpress/android/login/webauthn/PasskeyRequest.kt:
```diff
@@ -18,6 +19,7 @@ import org.wordpress.android.fluxc.generated.AuthenticationActionBuilder
 import org.wordpress.android.fluxc.store.AccountStore.FinishWebauthnChallengePayload
 import java.util.concurrent.Executors
 
+@SuppressLint("CredentialManagerMisuse")
```
@ParaskP7 I realize this is still a draft, but this suppression caught my eye and I wanted to point out we can remove it if we instead catch `NoCredentialException`below.

ParaskP7 replied:
👋 @nbradbury and thank you so much for taking a look at it, no matter if it is still a draft, much appreciated! 🙇 

About the suppression, and as per the description in the [commit](https://github.com/wordpress-mobile/WordPress-Android/pull/21680/commits/3559aa373ebf4214bf0cc6f1dda1eb728b63dc96) message, I first indeed tried to resolve the suppression, only to realize that `NoCredentialException` is actually a subclass of `GetCredentialException`, which means that it is already caught. 🤷 

Let me know what you think or if I am missing something obvious here... 🫣

nbradbury replied:
@ParaskP7 This may be a bug in lint but when I catch `GetCredentialException` without the suppression, lint flags it as an error.

But when I change it to catch `NoCredentialException` lint is just fine with it, despite it being a subclass of`GetCredentialException`. 🤷

ParaskP7 replied:
Exactly this @nbradbury , it smells like a bug to me too (ish), thus the so many 🤷 🤷 🤷 from both of us... 😅 

PS: The only reasoning I could think of is if someone would want a different action based on one or the other exception, like this below, where you need to have `NoCredentialException` first, otherwise it will be consumed by `GetCredentialException`:

```kotlin
        } try
            ...
        } catch (e: Exception) {
            when (e) {
                is NoCredentialException -> {
                    // Do something...
                }
                is GetCredentialException -> {
                    // Do something else...
                }
                else -> throw e
            }
        }
```

But, if you want to `Do something...` for both cases, the code as is, it is perfectly fine. 💯

ParaskP7 committed:
Deps: Update gutenberg kit version to pr hash

This 'Gutenberg Kit' PR hash updates the library to that branch
version where 'Gradle & AGP' is also upgraded to version
'8.12.1 & 8.8.0' respectively.

Gutenberg Kit PR: https://github.com/wordpress-mobile/
GutenbergKit/pull/92

This step is required in order to check that these 'Gutenberg Kit'
related changes work as expected for JP/WPAndroid.

ParaskP7 commented:
👋 @dcalhoun, about [this](https://github.com/wordpress-mobile/WordPress-Android/pull/21515#issuecomment-2650281240), I just wanted to share the progress done so far so that you know what to expect reviewing/supporting this work to completion next week.

1. The PR is ready to be reviewed/merged as is, without `local build` support.
2. The [GBKit#92](https://github.com/wordpress-mobile/GutenbergKit/pull/92) PR is also ready to be reviewed, and (if need be) worked-on, which will ultimately unblock `local build` support[^1] for GBKit. What I did with that:
    - I tested the standalone [GBKit#android-app](https://github.com/wordpress-mobile/GutenbergKit/tree/trunk/android/app) and it **seems** to be working as expected.
    - I tested JP/WPAndroid using [GBKit#92](https://github.com/wordpress-mobile/GutenbergKit/pull/92) PR as the `local build` and it **seems** to be working as expected.
    - I tested JP/WPAndroid using [GBKit#92#87368f3c253f5efa0abc4af05c7f3cd9e634ae3a](https://github.com/wordpress-mobile/GutenbergKit/pull/92/commits/87368f3c253f5efa0abc4af05c7f3cd9e634ae3a) as the `gutenberg-kit` version (619e7e3705d9c75f7ef3bdde84172e8a99699772) and it did work as expected.

[^1]: It would be good if also test these scenarios, just like I did, and then let me know if everything is working as expected, or more work is needed. Then, we could most probably be able to move forward, following the `Merge Instructions` that are mentioned on the PR's description, first merge [GBKit#92](https://github.com/wordpress-mobile/GutenbergKit/pull/92) PR, then `gutenberg-kit` version to reference the `trunk` hash or `tag` and finally review/merge this as a whole.

FYI: I'll now mark this as `Ready for review` and add you, @nbradbury and @wzieba as reviewers to move this forward. 🏁 

PS: Let me know if I am missing something in terms of how to progress this. With [GBM](https://github.com/wordpress-mobile/gutenberg-mobile), we followed a similar process, thus, I just copied that. Not sure if the same applies to [GBKit](https://github.com/wordpress-mobile/GutenbergKit) and to what extend. 🙏

nbradbury commented:
Results from my first round with this:

✅ Smoke test both apps without composite builds
❌ Smoke test both apps again using the composite builds, focusing solely on GBK 
✅ ./gradlew assembleJetpackVanillaRelease
✅ ./gradlew lintJetpackVanillaRelease

The composite build fails with, "Using multiple versions of the Android Gradle Plugin [8.8.0, 8.5.1] across Gradle builds is not allowed," but that's expected until the GBK update is ready.

ParaskP7 commented:
👋 @nbradbury and thanks for the review/testing! 🙇 

> ❌ Smoke test both apps again using the composite builds, focusing solely on GBK
> 
> The composite build fails with, "Using multiple versions of the Android Gradle Plugin [8.8.0, 8.5.1] across Gradle builds is not allowed," but that's expected until the GBK update is ready.

Actually, that's why I already created this [GBKit#92](https://github.com/wordpress-mobile/GutenbergKit/pull/92) PR too. Now, if you first checkout this PR locally, which will point this repo to `build/update-gradle-to-8.12.1-and-agp-to-8.8` as well, and then try smoke testing both apps again using the composite builds, you will notice that `Using multiple versions... not allowed` will go await, and this is because both repos will now point to `Gradle 8.12.2` and `AGP 8.8.0`.

Hope that helps resume you testing this path as well. 🙏

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into build/update-gradle-to-8.12.1-and-agp-to-8.8

ParaskP7 committed:
Deps: Update gutenberg kit version to pr hash (v2)

This 'Gutenberg Kit' PR hash updates the library to that branch
version where 'Gradle & AGP' is also upgraded to version
'8.12.1 & 8.8.0' respectively.

Gutenberg Kit PR: https://github.com/wordpress-mobile/
GutenbergKit/pull/92

This step is required in order to check that these 'Gutenberg Kit'
related changes work as expected for JP/WPAndroid.

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into build/update-gradle-to-8.12.1-and-agp-to-8.8

ParaskP7 committed:
Build: Upgrade agp to 8.8.1

Release Notes: https://developer.android.com/build/releases/
gradle-plugin

ParaskP7 committed:
Deps: Update gutenberg kit version to pr hash (v3)

This 'Gutenberg Kit' PR hash updates the library to that branch
version where 'Gradle & AGP' is also upgraded to version
'8.12.1 & 8.8.1' respectively.

Gutenberg Kit PR: https://github.com/wordpress-mobile/
GutenbergKit/pull/92

This step is required in order to check that these 'Gutenberg Kit'
related changes work as expected for JP/WPAndroid.

ParaskP7 commented:
FYI: I just updated `trunk` again but most importantly I included the new patch version of AGP ([8.8.1](62bf6a40f7edd117d0b42487e37b6d8f9be51db8)) and did the same for GBKit (828132634def55f739ccce634592adae1af356aa) so that when this PR is merged we get all the fixes it includes.

dcalhoun reviewed (commented):
Test plan succeeded for me aside from one `./gradlew lintJetpackVanillaRelease` error, but this appears to be present in `trunk` as well. So, I think it is safe to merge this. 

I'll let others more experienced in Android development provide formal PR approval. Thank you for completing this upgrade. 🙇🏻‍♂️ 

```
android/Gutenberg/src/main/java/org/wordpress/gutenberg/GutenbergView.kt:144: Error: The intent action android.intent.action.VIEW (used to start an activity) matches the intent filter of a non-exported component org.wordpress.android.ui.JetpackConnectionResultActivity from a manifest. If you are trying to invoke this specific component via the action then you should make the intent explicit by calling Intent.set{Component,Class,ClassName}. [UnsafeImplicitIntentLaunch]
                      val intent = Intent(Intent.ACTION_VIEW, Uri.parse(it))
                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

ParaskP7 commented:
> Test plan succeeded for me aside from one ./gradlew lintJetpackVanillaRelease error, but this appears to be present in trunk as well. So, I think it is safe to merge this.

Awesome @dcalhoun , thanks for testing and verifying that everything is working as expected! 🙇 🎉 🚀 

PS: Yea, about the  `./gradlew lintJetpackVanillaRelease` error with the GBKit related `android/Gutenberg/src/main/java/org/wordpress/gutenberg/GutenbergView.kt` file:
1. As you said, as long as this was already the case on `trunk` I wouldn't worry about it.
2. Since `./gradle lintRelease` on [GBKit#92](https://github.com/wordpress-mobile/GutenbergKit/pull/92) succeeds we should be fine.
3. Since `./gradle lintJetpackVanillaRelease` without enabling the `localGutenbergKitPath` local build succeeds we should be fine.
4. Having the `./gradle lintJetpackVanillaRelease` erroring when the `localGutenbergKitPath` local build is enabled  (IMHO) should be fixed, but this doesn't (necessarily) need to be part of this PR.

> I'll let others more experienced in Android development provide formal PR approval. Thank you for completing this upgrade. 🙇🏻‍♂️

Great, so, after we get a 👍 on this from @nbradbury (and maybe also @wzieba) , we can go ahead and merge this [GBKit#92](https://github.com/wordpress-mobile/GutenbergKit/pull/92) PR and more forward with the rest of the `Merge Instructions` steps, right? 🙏

dcalhoun commented:
> Great, so, after we get a 👍 on this from @nbradbury (and maybe also @wzieba) , we can go ahead and merge this [GBKit#92](https://github.com/wordpress-mobile/GutenbergKit/pull/92) PR and more forward with the rest of the `Merge Instructions` steps, right? 

Correct.

nbradbury reviewed (approved):
@ParaskP7 This is working great for me! I'll approve it, and you can merge either now or after @wzieba takes a look. :shipit:

ParaskP7 commented:
Awesome @nbradbury , thanks for taking a final look at it! 🙇❤️🚀

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into build/update-gradle-to-8.12.1-and-agp-to-8.8

ParaskP7 commented:
Awesome folks, I got all the 👍s I needed to proceed with the rest of the merge instructions, I am currently on it, starting with merging [GBKit#92](https://github.com/wordpress-mobile/GutenbergKit/pull/92), thank YOU all! 🙇 🏃

ParaskP7 committed:
Deps: Update gutenberg kit version to trunk hash

This 'Gutenberg Kit' trunk hash updates the library to that branch
version where 'Gradle & AGP' is also upgraded to version
'8.12.1 & 8.8.1' respectively.

Gutenberg Kit PR: https://github.com/wordpress-mobile/
GutenbergKit/pull/92

This step is required in order to check that these 'Gutenberg Kit'
related changes work as expected for JP/WPAndroid.
